easybuild/easyblocks/x/xmipp.py

```diff
@@ -0,0 +1,161 @@
+        # strip off 'xmipp' part to avoid having everything in a 'xmipp' subdirectory
+        if not self.cfg['unpack_options']:
+            self.cfg['unpack_options'] = '--strip-components=1'
+        super(EB_Xmipp, self).extract_step()
+
+    def configure_step(self):
+        """Configure xmipp build via a provided wrapper around scons."""
+        # check if all our dependencies are in place
+        python_root = get_software_root('Python')
+        if not python_root:
+            self.log.error("Python not loaded as a dependency, which is required for %s" % self.name)
+        python_libdir = det_pylibdir()
+        self.python_short_ver = ".".join(get_software_version('Python').split(".")[0:2])
+        java_root = get_software_root('Java')
+        if not java_root:
+            self.log.error("Java not loaded as a dependency, which is required for %s" % self.name)
+
+        # extract some dependencies that we really need and can't find anywhere else
+        # alglib tarball has version in name, so lets find it with a glob
+        # we can't do this in extract step before these are in the original sources tarball, so we need to know
+        # startdir first
+        external_path = os.path.join(self.cfg['start_dir'], 'external')
+        alglib_tar = glob.glob(os.path.join(external_path, 'alglib*.tgz'))[0]
+        for src in ['bilib.tgz', 'bilib.tgz', 'condor.tgz', alglib_tar, 'scons.tgz']:
+            extract_file(os.path.join(external_path, src), external_path)
+
+        # make sure we start in the start dir
+        os.chdir(self.cfg['start_dir'])
+
+        # build step expects these to exist
+        mkdir(os.path.join(self.cfg['start_dir'], 'bin'))
+        mkdir(os.path.join(self.cfg['start_dir'], 'lib'))
+
+        cmd = ' '.join([
+            self.cfg['preconfigopts'],
+            'python external/scons/scons.py',
+            'mode=configure',
+            '-j {parallel}',
+            '--config=force',
+            'profile=no',
+            'fast=yes',
+            'warn=no',
+            'release=yes',
+            'gtest=no',
+            'cuda=no',
+            'debug=no',
+            'matlab=no',
+            'java=no',
+            'LINKERFORPROGRAMS="$CXX"',
+            'MPI_BINDIR="{mpi_bindir}"',
+            'JAVA_HOME="{java_home}"',
+            'JAVAC=javac',
+            'CC="$CC"',
+            'CXXFLAGS="%s"' % ' '.join([
+                '$CXXFLAGS',
+                '-DMPICH_IGNORE_CXX_SEEK',
+                '-I$EBROOTPYTHON/include/python{short_python_ver}',
+                '-I$EBROOTPYTHON/{python_libdir}/numpy/core/include/',
+            ]),
+            'CXX="$CXX"',
+            'MPI_CC="$MPICC"',
+            'MPI_CXX="$MPICXX"',
+            'MPI_INCLUDE="$MPI_INC_DIR"',
+            'MPI_LIBDIR="$MPI_LIB_DIR"',
+            'MPI_LINKERFORPROGRAMS="$MPICC"',
+            'LIBPATH="$LD_LIBRARY_PATH"',
+            self.cfg['configopts'],
+        ]).format(
```

Comment:
well, it works like this, so I'll leave it for now. 
combined with the step below this can be a nice example for people to see the 2 ways you can do string templating in python, and how the new .format function works, so they can get ready to start using it.